service/device/registry/test/src/test/resources/features/DeviceRegistry.feature

```diff
@@ -16,237 +16,209 @@ Feature: Device Registry CRUD tests
+        Empty list should be returned.
+
+        When I search for a device with a random ID
+        Then There is no such device
+
+    Scenario: Try to find a device with an invalid client ID
+        Searching for a nonexistent device should not raise any exception.
+        Empty list should be returned.
+
+        When I search for a device with a random client ID
+        Then There is no such device
+
+    Scenario: Device query - find by BIOS version
+        It must be possible to construct arbitrary device registry queries. In this case
+        several test devices are created with different BIOS version values.
+        A query based on the BIOS version must only return the correct device.
+
+        Given A device with BIOS version "1.1.0" named "TestDevice1"
+        Given A device with BIOS version "1.2.0" named "TestDevice2"
+        Given A device with BIOS version "1.3.0" named "TestDevice3"
+        When I query for devices with BIOS version "1.1.0"
+        And I extract the first device
+        Then The device client id is "TestDevice1"
+
+    Scenario: Device queries
+        Test several variants of device registry queries.
+        Kapua should not return any error.
+
+        Given A device named "TestDevice"
+        Given I create 100 randomly named devices with BIOS version "1.1.0"
+        Given I create 201 randomly named devices with BIOS version "1.2.0"
+        Given I create 98 randomly named devices with BIOS version "1.3.0"
+        When I query for devices with BIOS version "1.1.0"
+        Then I find 100 devices
+        When I query for devices with BIOS version "1.3.0"
+        Then I find 98 devices
+        When I query for devices with BIOS different from "1.2.0"
+        Then I find 201 devices
+        When I query for devices with Client Id "TestDevice"
+        Then I find 1 device
+
+    Scenario: Count devices with a specific BIOS version
+        It must be possible to count devices based on arbitrary rules.
+        To this end several devices are created with different BIOS version.
+        A device count based on the BIOS version value is performed. Only
+        the number of devices that match the specified BIOS version is
+        returned.
+
+        Given I create 15 randomly named devices with BIOS version "1.1.0"
+        Given I create 25 randomly named devices with BIOS version "1.2.0"
+        Given I create 35 randomly named devices with BIOS version "1.3.0"
+        When I count devices with BIOS version "1.2.0"
+        Then I count 25
+
+    Scenario: Try to update the device client ID
+        The Client ID of a device cannot be changed after creation. Any attempt to
+        after this ID must be silently ignored. No exception should be raised.
+
+        Given A device named "TestDevice"
+        When I update the device clientID from "TestDevice" to "NewClientId"
+        Then The client ID was not changed
```

Comment:
Please change or reuse one of the previous steps ("Then I tryi to find device with ID.."). Just checking in the background if this has been done or not is not enough.